components/raftstore/src/store/entry_storage.rs

```diff
@@ -479,12 +479,8 @@ fn validate_states<ER: RaftEngine>(
         info!("updating commit index"; "region_id" => region_id, "old" => commit_index, "new" => recorded_commit_index);
         commit_index = recorded_commit_index;
     }
-    // Invariant: applied index <= max(commit index, recorded commit index)
     if apply_state.get_applied_index() > commit_index {
-        return Err(box_err!(
-            "applied index > max(commit index, recorded commit index), {}",
-            state_str()
-        ));
+        info!("applied index is larger than recorded commit index"; "apply" => apply_state.get_applied_index(), "commit" => commit_index);
```

Comment:
Do we need to log it anymore?  I think maybe just using a metrics to indicate how frequently it happens.

Reply:
There is a metric that count the number of logs that applied_index is ahead of persisted_index. I think we better keep this log since it is different from the current logic and we don't expect it happens to too many regions.